docusaurus/docs/common-use-cases/providing-viewmodels.md

```diff
@@ -211,9 +211,53 @@ class ViewModelFactory(
 }
 ```
 
-As seen above, if you want a single function for all viewModel types, then you need a single `ViewModelFactory` that can create all your types of ViewModels. Alternatively, you could have one function for each ViewModel (example: `profileViewModel()` to get `ProfileViewModel` by using a `ProfileViewModelFactory`).
+As seen above, if you want a single function for all viewModel types, then you need a single `ViewModelFactory` that can create all your types of ViewModels. This does mean that every time you add a new ViewModel to your code base, you'll have to come here and add that entry to the `when` statement above. Nothing is perfect, and this is definitely one reason to make you want to use a Dependency Injection framework like Hilt, so that this is handled for you.
```

Comment:
> This does mean that every time you add a new ViewModel to your code base, you'll have to come here and add that entry to the `when` statement above. Nothing is perfect, and this is definitely one reason to make you want to use a Dependency Injection framework like Hilt, so that this is handled for you.

I think this paragraph should be below the alternative approach too, maybe change it slightly. In the end, this is true for both approaches. One you have to add an entry to the when statement, the other you have to create a new `viewModel` getter function 🙂